public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryEditorSelector.test.tsx

```diff
@@ -150,23 +168,31 @@ function renderWithMode(mode: QueryEditorMode) {
   return renderWithProps({ editorMode: mode } as any);
 }
 
-function renderWithProps(overrides?: Partial<PromQuery>) {
+function renderWithProps(overrides?: Partial<PromQuery>, componentProps: Partial<PromQueryEditorProps> = {}) {
   const query = defaultsDeep(overrides ?? {}, cloneDeep(defaultQuery));
   const onChange = jest.fn();
 
-  const stuff = render(<PromQueryEditorSelector {...defaultProps} query={query} onChange={onChange} />);
+  const allProps = { ...defaultProps, ...componentProps };
+  const stuff = render(<PromQueryEditorSelector {...allProps} query={query} onChange={onChange} />);
   return { onChange, ...stuff };
 }
 
 function expectCodeEditor() {
-  // Metric browser shows this until metrics are loaded.
-  expect(screen.getByText('Loading metrics...')).toBeInTheDocument();
+  expect(screen.getByText('MonacoQueryFieldWrapper')).toBeInTheDocument();
 }
 
 function expectBuilder() {
   expect(screen.getByText('Metric')).toBeInTheDocument();
 }
 
+function expectRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).toBeInTheDocument();
+}
+
+function expectNoRunQueriesButton() {
+  expect(screen.queryByText('Run queries')).not.toBeInTheDocument();
+}
+
```

Comment:
same as the other comment:
```suggestion
let's use more semantic selectors:
```suggestion
function expectRunQueriesButton() {
  expect(screen.getByRole('button', { name: /run queries/i })).toBeInTheDocument();
}

function expectNoRunQueriesButton() {
  expect(screen.queryByRole('button', { name: /run queries/i })).not.toBeInTheDocument();
}
```
```